CourseSystem/BLL/Services/AssignmentAnswerService.cs

```diff
@@ -1,35 +1,46 @@
-﻿using BLL.Interfaces;
+﻿using System.Reflection;
+using BLL.Interfaces;
 using Core.Enums;
 using Core.Models;
 using DAL.Repository;
 using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
 
 namespace BLL.Services
 {
     public class AssignmentAnswerService : GenericService<AssignmentAnswer>, IAssignmentAnswerService
     {
         private readonly IUserAssignmentService _userAssignmentService;
         private readonly IDropboxService _dropboxService;
+        private readonly ILogger<AssignmentAnswerService> _logger;
 
-        public AssignmentAnswerService(UnitOfWork unitOfWork, IUserAssignmentService userAssignmentService, IDropboxService dropboxService)
+        public AssignmentAnswerService(UnitOfWork unitOfWork, IUserAssignmentService userAssignmentService, IDropboxService dropboxService, ILogger<AssignmentAnswerService> logger)
             : base(unitOfWork, unitOfWork.AssignmentAnswerRepository)
         {
             _userAssignmentService = userAssignmentService;
             _dropboxService = dropboxService;
+            _logger = logger;
         }
 
         public async Task<Result<bool>> CreateAssignmentAnswer(AssignmentAnswer assignmentAnswer, Assignment assignment,
             AppUser appUser, IFormFile? file, DropboxFolders folder)
         {
             if (assignmentAnswer == null)
             {
+                _logger.LogError("Failed to {action}, {entity} was null!", MethodBase.GetCurrentMethod()?.Name, nameof(assignmentAnswer));
+
                 return new Result<bool>(false, "Invalid assignment answer");
             }
 
+            _logger.LogInformation("Forming {action} for user {userId} about assignmentAnswer {assignmentId}",
```

Comment:
Strange wording ("Forming assignmentAnswer for user {userId} and/about assignment {assignmentId}")

Reply:
Fixed